Packages/com.unity.inputsystem/InputSystem/InputManager.cs

```diff
@@ -2648,17 +2658,27 @@ internal void AddAvailableDevicesThatAreNowRecognized()
                     continue;
 
                 var layout = TryFindMatchingControlLayout(ref m_AvailableDevices[i].description, id);
-                if (IsDeviceLayoutMarkedAsSupportedInSettings(layout))
+                if (!IsDeviceLayoutMarkedAsSupportedInSettings(layout)) continue;
+
+                if (layout.IsEmpty())
                 {
-                    try
-                    {
-                        AddDevice(m_AvailableDevices[i].description, false,
-                            deviceId: id,
-                            deviceFlags: m_AvailableDevices[i].isNative ? InputDevice.DeviceFlags.Native : 0);
-                    }
-                    catch (Exception)
+                    // If it's a device coming from the runtime, disable it.
+                    if (id != InputDevice.InvalidDeviceId)
                     {
+                        var command = DisableDeviceCommand.Create();
+                        m_Runtime.DeviceCommand(id, ref command);
                     }
+
+                    continue;
+                }
+
+                try
+                {
+                    AddDevice(m_AvailableDevices[i].description, layout, deviceId: id,
+                        deviceFlags: m_AvailableDevices[i].isNative ? InputDevice.DeviceFlags.Native : 0);
+                }
+                catch (Exception)
```

Comment:
Is this error commonly excepted or do we want to see it in e.g. Debug log at least? If intended to suppress maybe add a comment why we ignore the error?

Reply:
Wasn't sure, so left as is. Maybe @Rene-Damm knows?

Reply:
Comment probably a good idea. Reason we just suppress the exception is that here in the `AddAvailableDevicesThatAreNowRecognized` method we're doing a kind of "you know, our layout stuff has changed so maybe we can make sense of some stuff now what we couldn't make sense of before... let's try". But that trying may still fail. So, thinking is, if you make a completely unrelated change to the layout information, you shouldn't get spammed with errors from an unrelated layout that happens to be cause issues.

Reply:
@andrew-oc Could we add a one-liner comment just explaining the why along these lines for future reference? An empty catch without explanation looks like "forgetting to add handling code"